module/reports/helper.js

```diff
@@ -0,0 +1,443 @@
+                            }
+                        });
+                    }
+
+                    await Promise.all(project.tasks.map(task => {
+                        var todayDate = moment(task.endDate, "DD.MM.YYYY");
+                        var endDate = moment().format();
+                        if (todayDate.diff(endDate, 'days') < 1) {
+                            if (tasksReport['overdue']) {
+                                tasksReport['overdue'] = tasksReport['overdue'] + 1;
+                            } else {
+                                tasksReport['overdue'] = 1;
+                            }
+                        }
+                    }));
+
+
+                }));
+
+                if (projectReport[CONSTANTS.common.INPROGRESS_STATUS]) {
+                    projectReport['onGoing'] = projectReport[CONSTANTS.common.INPROGRESS_STATUS];
+                    delete projectReport['inProgress'];
+                }
+
+                if (tasksReport[CONSTANTS.common.INPROGRESS_STATUS]) {
+                    tasksReport['onGoing'] = tasksReport[CONSTANTS.common.INPROGRESS_STATUS];
+                    delete tasksReport[CONSTANTS.common.INPROGRESS_STATUS];
+                }
+
+                let response = {
+                    categories: categories,
+                    tasks: tasksReport,
+                    projects: projectReport
+                }
+                return resolve({
+                    message: CONSTANTS.apiResponses.REPORTS_GENERATED,
+                    result: {
+                        data_available:true,
+                        data:response,   
+                    }
+                });
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+
+    /**
+      * Get programs list.
+      * @method
+      * @name getProgramsByEntity 
+      * @param {String} userId - Logged in user id.
+      * @param {String} entityId - entity id.
+      * @param {String} pageSize - Size of page.
+      * @param {String} pageNo - Recent page no.
+      * @param {String} search - search text.
+      * @returns {Object} -  returns programs list
+     */
+
+    static getProgramsByEntity(userId, entityId, pageSize, pageNo, search) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let query = {
+                    "entityInformation._id": mongoose.Types.ObjectId(entityId),
+                }
+
+                let searchQuery = [];
+                if (search !== "") {
+                    searchQuery = [{ "programInformation.name": new RegExp(search, 'i') }];
+                }
+
+                const projectDocuments = await UserProjectsHelper.projects(
+                    query,
+                    pageSize,
+                    pageNo,
+                    searchQuery,
+                    ["programInformation", "entityInformation", "userId"]
```

Comment:
@rakeshSgr - Why do we need entityInformation ?

Reply:
@rakeshSgr - Only project program name

Reply:
resolved